### PR TITLE
[WIP] [JavaScript] JavaScript Standard Library Declarations

### PR DESCRIPTION
**Pull request by harlanhaskins**
This PR tracks implementations of foreign declarations of JavaScript standard library types.

### CODE REVIEW

File: String.tr [hunk 1]
```diff
@@ -0,0 +1,38 @@
+foreign type String {
+foreign var length: Int
```
Comment (segiddins) [inline String.tr]: indent

File: String.tr [hunk 2]
```diff
@@ -0,0 +1,38 @@
+foreign type String {
```
Comment (segiddins) [inline String.tr]: should be nested in the new directory

**Review by segiddins — COMMENTED** [review]
Do you think this is a better approach than either automatic importing from the JS runtime, or some sort of conditional compilation in the existing stdlib trill files?

**Comment by harlanhaskins** [comment]
There's no way to automatically import. We need types.

File: trill.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1336,6 +1341,13 @@
 				DCAE21EC1E63970A00645143 /* main.swift */,
 			);
 			path = "trill-demangle";
+		DC8EBCCC1E6366A400AE577C /* JavaScript Declarations */ = {
```
Comment (segiddins) [inline trill.xcodeproj/project.pbxproj]: bad merge here?

**Comment by harlanhaskins** [comment]
The JavaScript backend just can't be unified with the LLVM backend in its current state. I'd rather them be completely separate if they can't be completely unified.

**Comment by harlanhaskins** [comment]
That said, I'd love to use something like [emscripten](http://emscripten.org) to generate asm.js and run that instead.

**Comment by terhechte** [comment]
I'm just a lurker on here, but the verbose Javascript Generation is actually something I really like about Trill. On the server, Swift can already be used fairly comfortably. I'd love to be able to replace client code with something as close to Swift as possible. One can always run the output through Google's Closure library in order to gain dead code elimination much like ClojureScript does it.

**Comment by harlanhaskins** [comment]
My issue with the verbose JavaScript output is that it really doesn't reflect the semantics of the language...

Though I'm cool considering it a separate dialect of Trill with separate semantics. I'd just shy away from recommending anyone use Trill for anything serious 😅

**Pull request closed without merging**